src/builtins/ulimit.rs

```diff
@@ -70,11 +70,22 @@ pub mod limits {
     pub mod bsd {
         use libc;
 
+        #[cfg(any(target_os = "dragonfly", target_os = "freebsd", target_os = "netbsd"))]
         pub const SBSIZE: libc::c_int = libc::RLIMIT_SBSIZE;
         pub const RSS: libc::c_int = libc::RLIMIT_RSS;
         pub const SWAP: libc::c_int = libc::RLIMIT_SWAP;
+        #[cfg(not(target_os = "freebsd"))]
+        pub const SWAP: libc::c_int = -1;
+
+        #[cfg(target_os = "freebsd")]
         pub const KQUEUES: libc::c_int = libc::RLIMIT_KQUEUES;
+        #[cfg(not(target_os = "freebsd"))]
+        pub const KQUEUES: libc::c_int = -1;
```

Comment:
Can you describe what this fixes, and what you tested?
AFAICT it doesn't fix OpenBSD,
e.g. https://github.com/fish-shell/fish-shell/issues/11892
mentions the lack of `RLIMIT_AS`

Please make a PR adding that to `rust-lang/libc` first;
then we can pull the new version